### PR TITLE
[Streaming] Streaming scheduler - part1-1: job graph

### PR DESCRIPTION
**Pull request by BalaBalaYi**
## Why are these changes needed?

1. Rename ```Plan``` to ```JobGraph```.
2. Add ``jobName`` as essential parameter in ```JobGraph``` and ```StreamContext```.
3. Change config type from ```Map<String, Object>``` to ```Map<String, String>```.
notice: The string type config value will convert to specific type by owner config automatically.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by chaokunyang — COMMENTED** [review]
Rename `Plan` to `JobGraph`

File: streaming/java/streaming-api/src/main/java/org/ray/streaming/api/context/StreamingContext.java [hunk 1]
```diff
@@ -24,11 +24,11 @@
    * The sinks of this streaming job.
    */
   private List<StreamSink> streamSinks;
-  private Map<String, Object> jobConfig;
+  private Map<String, String> jobConfig;
```
[the remark this answers is not in the document]
Comment (BalaBalaYi) [reply streaming/java/streaming-api/src/main/java/org/ray/streaming/api/context/StreamingContext.java]: Job configuration info only.

File: streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  public List<JobVertex> getJobVertexList() {
+    return jobVertexList;
+  }
+
+  public List<JobEdge> getJobEdgeList() {
+    return jobEdgeList;
+  }
+
+  public String getGraphViz() {
+    return "";
```
Comment (chaokunyang) [inline streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: Why return an empty string? Can we return a string which can be visualized by `graphviz`?

Comment (BalaBalaYi) [reply streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: This used to be:
image

> Can we return a digraph string which can be visualized by `graphviz`?
Will do.

File: streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    this.jobVertexList = new ArrayList<>();
+    this.jobEdgeList = new ArrayList<>();
+  }
+
+  /**
+   * Generate direct graph by current job graph.
+   * @return Digraph in string type.
+   *
+   * Notice:
+   * This is a temporarily implement in hard code.
```
Comment (raulchen) [inline streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]:
```suggestion
   * This is temporarily implemented in hard code.
```

**Review by raulchen — COMMENTED** [review]
LGTM. 
@chaokunyang @JingGe @liangyi123 do you have other comments?

File: streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+   * May use 'guru.nidi:graphviz-java' in the future if needed.
+   */
+  public String generateDigraph() {
```
Comment (liangyi123) [inline streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: we‘d better explain more about what is digraph and why it is needed in javadoc for this method or this class

Comment (BalaBalaYi) [reply streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: Optimized.

Comment (liangyi123) [reply streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: Looks good now.

File: streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+   * Notice:
+   * This is temporarily implemented in hard code.
+   * May use 'guru.nidi:graphviz-java' as 3rd dependency in the future if needed.
+   */
+  public String generateDigraph() {
+    StringBuilder digraph = new StringBuilder();
+    digraph.append("digraph ").append(jobName + " ").append(" {");
+
+    jobEdgeList.stream().forEach(jobEdge -> {
+      AtomicReference<String> srcNode = new AtomicReference<>();
```
Comment (chaokunyang) [inline streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: Why use `AtomicReference`, there is no multi-thread here. If i understood right, you want use a reference in lambda? Why not just use loop?

Comment (BalaBalaYi) [reply streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraph.java]: Use loop instead of lambda.

**Review by chaokunyang — APPROVED** [review]
LGTM

**Review by raulchen — APPROVED** [review]

**Pull request merged**